ignite/contrib/handlers/trains_logger.py

```diff
@@ -0,0 +1,636 @@
+
+            # Attach the logger to the evaluator on the validation dataset and log NLL, Accuracy metrics after
+            # each epoch. We setup `global_step_transform=global_step_from_engine(trainer)` to take the epoch of the
+            # `trainer` instead of `evaluator`.
+            trains_logger.attach(evaluator,
+                                 log_handler=OutputHandler(tag="validation",
+                                                          metric_names=["nll", "accuracy"],
+                                                          global_step_transform=global_step_from_engine(trainer)),
+                                 event_name=Events.EPOCH_COMPLETED)
+
+            # Attach the logger to the trainer to log optimizer's parameters, e.g. learning rate at each iteration
+            trains_logger.attach(trainer,
+                                 log_handler=OptimizerParamsHandler(optimizer),
+                                 event_name=Events.ITERATION_STARTED)
+
+            # Attach the logger to the trainer to log model's weights norm after each iteration
+            trains_logger.attach(trainer,
+                                 log_handler=WeightsScalarHandler(model),
+                                 event_name=Events.ITERATION_COMPLETED)
+
+
+    """
+
+    _bypass = None
+
+    def __init__(self, *_, **kwargs):
+        try:
+            import trains
+        except ImportError:
+            raise RuntimeError(
+                "This contrib module requires trains to be installed. "
+                "You may install trains using: \n pip install trains \n"
+            )
+
+        experiment_kwargs = {
+            k: v
+            for k, v in kwargs.items()
+            if k
+            not in (
+                "project_name",
+                "task_name",
+                "task_type",
+                "report_freq",
+                "histogram_update_freq_multiplier",
+                "histogram_granularity",
+            )
+        }
+
+        if self.bypass_mode():
+            warnings.warn("TrainsSaver: running in bypass mode")
+
+            class _Stub(object):
+                def __call__(self, *_, **__):
+                    return self
+
+                def __getattr__(self, attr):
+                    if attr in ("name", "id"):
+                        return ""
+                    return self
+
+                def __setattr__(self, attr, val):
+                    pass
+
+            self._task = _Stub()
+        else:
+            self._task = trains.Task.init(
+                project_name=kwargs.get("project_name"),
+                task_name=kwargs.get("task_name"),
+                task_type=kwargs.get("task_type", trains.Task.TaskTypes.training),
+                **experiment_kwargs,
+            )
+
+        self.trains_logger = self._task.get_logger()
+
+        self.grad_helper = trains.binding.frameworks.tensorflow_bind.WeightsGradientHistHelper(
```

Comment:
I'm trying to find this class (`WeightsGradientHistHelper`) in your codebase: https://github.com/allegroai/trains/blob/master/trains/binding/frameworks/tensorflow_bind.py
@jkhenning is it related to your rc version ?

Reply:
This is exactly the small refactoring in the RC version 😄

Reply:
Okay, thanks ! Just a bit funny to see tensorflow term in pytorch based code :)